spark-jobs/src/main/scala/za/co/absa/enceladus/common/CommonJobExecution.scala

```diff
@@ -142,8 +166,42 @@ trait CommonJobExecution {
     }
   }
 
+  protected def getPathConfig[T](cmd: JobConfigParser[T], dataset: Dataset, reportVersion: Int): PathConfig = {
+    PathConfig(
+      rawPath = buildRawPath(cmd.asInstanceOf[StandardizationConfigParser[StandardizationConformanceConfig]], dataset, reportVersion),
+      publishPath = buildPublishPath(cmd.asInstanceOf[ConformanceConfigParser[StandardizationConformanceConfig]], dataset, reportVersion),
```

Comment:
Nice solution
Would not cast the `cmd` here. Let the function within `CommonJobExecution` here return only the default values. And in the overrides check the class, cast it, and eventually use the configuration override.